### ISSUE TITLE
How to produce 68 face landmark points?

### ISSUE THREAD

**Issue by herleeyandi** [kpzhang93/MTCNN_face_detection_alignment#13]
Hello I am new in face detection and landmark detection. My question is based on this network how can I produce 68 points of landmark, because in thisdemo it just can produce 5 points. Do I need to train again with 68 landmark?
-Thank you-

**Comment by jianchao-li** [comment]
Yes, you need to train on labeled data with 68 facial landmarks. If you would like to do so, you need to change the `num_output` of `conv6-3` in `det3.prototxt` to be `136` (`2 * 68`)

**Comment by herleeyandi** [comment]
Hi thank you for your reply @jianchao-li . Could you give us the detail how to train the P-Net, R-Net, and O-Net?, they should train separately right? Also what is the training data format? for example O-Net needs: ROI of face [x1,y1,w,h] and also 5 landmarks (10 element in total). Where can we find the data and do you have any specific suggestion to prepare the data before training all of them?
-Thank you-

**Comment by jianchao-li** [comment]
Yeah, the three nets are trained separately, or more precisely, sequentially, which means that the latter net is trained based on the output of the former net.

For the preparation of training data, you may take a look at [mtcnn-caffe](https://github.com/CongWeilin/mtcnn-caffe).

**Comment by Andrawzyf** [comment]
@herleeyandi have you produce 68 face landmark points successfully?

**Comment by ZouaghiHoussem** [comment]
@herleeyandi did you manage to do it ? i'm in the same stage and  looking how to change it to 68pt
